### PR TITLE
kvm: truncate vnc password to 8 chars

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR truncates the vnc password of kvm vms to 8 chars.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by weizhouapache** [comment]
this requires manually testing on ubuntu 22.04 or libvirt 8. I will not sure if novnc console works

**Review by nvazquez — APPROVED** [review]

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
tested on centos7 and ubuntu20, vm console works.

I have not tested on ubuntu 22.04.

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtVMDef.java [hunk 1]
```diff
@@ -1764,7 +1764,7 @@ public String toString() {
                 graphicBuilder.append(" listen=''");
             }
             if (_passwd != null) {
-                graphicBuilder.append(" passwd='" + _passwd + "'");
+                graphicBuilder.append(" passwd='" + StringUtils.truncate(_passwd, 8) + "'");
```
Comment (rohityadavcloud) [inline plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtVMDef.java:1767]: Why is that @weizhouapache ? Does this fix/address any libvirt limitation or user reported issue?

Comment (weizhouapache) [reply plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtVMDef.java:1767]: @rohityadavcloud 
this aims to fix an issue with ubuntu 22.04 (to be released soon) and libvirt 8.
the issue is raised in mailing list: https://lists.apache.org/thread/zk4fyggjqk1zphk4dfg1m596yrd15d9b

from many years ago, libvirt accepts vnc password with more than 8 chars. but only the first 8 chars in vnc password are used in vnc console verification (the other chars are dropped).
In recent libvirt, only vnc password with 8 chars is permitted.

**Comment by weizhouapache** [comment]
@blueorangutan package

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtVMDef.java (hunk 1 above)
Comment (rohityadavcloud) [reply plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtVMDef.java:1767]: LGTM

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
@blueorangutan package

**Review by rohityadavcloud — APPROVED** [review]

**Pull request merged**